pkg/executables/govc.go

```diff
@@ -326,36 +337,76 @@ func (g *Govc) DeployTemplateFromLibrary(ctx context.Context, templateDir, templ
 		diskName := ""
 		var diskSizeInGB int
 		if disk2 != "" {
-			logger.V(4).Info("Resizing disk 2 of template to 20G")
+			logger.V(2).Info("Resizing disk 2 of template to 20G")
 			diskName = disk2
 			diskSizeInGB = 20
 		} else if disk1 != "" {
-			logger.V(4).Info("Resizing disk 1 of template to 22G")
+			logger.V(2).Info("Resizing disk 1 of template to 22G")
 			diskName = disk1
 			diskSizeInGB = 22
 		} else {
 			return fmt.Errorf("template %v is not valid as there are no associated disks", templateName)
 		}
 
+		logger.V(2).Info("Template Name. ", "Template name: ", templateName)
+		logger.V(2).Info("Printing diskSize for resize: ", "Disk Size: ", diskSizeInGB)
+
 		err = g.ResizeDisk(ctx, datacenter, templateName, diskName, diskSizeInGB)
 		if err != nil {
 			return fmt.Errorf("resizing disk %v to 20G: %v", diskName, err)
 		}
-	}
 
-	templateFullPath := filepath.Join(templateDir, templateName)
+		devicesInfo2, err := g.DevicesInfo(ctx, datacenter, templateName)
+		if err != nil {
+			return err
+		}
 
-	logger.V(4).Info("Taking template snapshot", "templateName", templateFullPath)
-	if err := g.createVMSnapshot(ctx, datacenter, templateFullPath); err != nil {
-		return err
+		for _, deviceInfo := range devicesInfo2.([]interface{}) {
+			deviceMetadata := deviceInfo.(map[string]interface{})["DeviceInfo"]
+			deviceLabel := deviceMetadata.(map[string]interface{})["Label"].(string)
+			deviceDiskSize := deviceMetadata.(map[string]interface{})["Summary"].(string)
+
+			if strings.EqualFold(deviceLabel, "Hard disk 1") || strings.EqualFold(deviceLabel, "Hard disk 2") {
+				parsedDiskSize := parseDiskSize(deviceDiskSize)
+
+				if parsedDiskSize == -1 {
+					logger.V(2).Info("Unable to get converted KB to GB int")
+					return fmt.Errorf("unable to get converted KB to GB int: %d", parsedDiskSize)
+				}
+				if disk2 != "" {
+					//check if disk1 is 20gb
+					if parsedDiskSize != 20 {
+						logger.V(2).Info("Unable to resize disk1 to 20 GB")
+						return fmt.Errorf("error resizing disk1 to 20 GB - new size is: %d", parsedDiskSize)
```

Comment:
As a general standard, we avoid having the word "error" in fmt.Errorfs. This would anyways print "Error: <message>". The fmt error above this is a good example. This could be `resizing disk1 to 20 GB - actual size is: %d`